tapsend/send_test.go

```diff
@@ -2952,3 +2952,151 @@ func TestValidateAnchorInputs(t *testing.T) {
 		})
 	}
 }
+
+// TestValidateScriptKeysUnique tests that the script keys in a set of virtual
+// packets are unique per TAP commitment key.
+func TestValidateScriptKeysUnique(t *testing.T) {
+	groupPubKey1 := test.RandPubKey(t)
+	groupPubKey2 := test.RandPubKey(t)
+	groupKey1 := &asset.GroupKey{
+		GroupPubKey: *groupPubKey1,
+	}
+	groupKey2 := &asset.GroupKey{
+		GroupPubKey: *groupPubKey2,
+	}
+	assetID1 := asset.ID{1, 2, 3}
+	assetID2 := asset.ID{2, 3, 4}
+
+	scriptKey1 := asset.NewScriptKey(test.RandPubKey(t))
+	scriptKey2 := asset.NewScriptKey(test.RandPubKey(t))
+	scriptKey3 := asset.NewScriptKey(test.RandPubKey(t))
+
+	makeVPacket := func(assetID asset.ID, groupKey *asset.GroupKey,
+		outputKeys []asset.ScriptKey) *tappsbt.VPacket {
+
+		vPkt := &tappsbt.VPacket{
+			ChainParams: &address.RegressionNetTap,
+			Inputs: []*tappsbt.VInput{
+				{
+					PrevID: asset.PrevID{
+						ID: assetID,
+					},
+				},
+			},
+			Outputs: make([]*tappsbt.VOutput, len(outputKeys)),
+		}
+		var a asset.Asset
+		if groupKey != nil {
+			a.GroupKey = groupKey
+		}
+		vPkt.SetInputAsset(0, &a)
+
+		for i, outputKey := range outputKeys {
+			vPkt.Outputs[i] = &tappsbt.VOutput{
```

Comment:
Shouldn't the output index be considered here when determining uniqueness?

Reply:
No, that's kind of the point of this fix. See comment(s) above.